frappe/model/mapper.py

```diff
@@ -282,3 +282,12 @@ def map_child_doc(source_d, target_parent, table_map, source_parent=None, target
 		target_parent.append(target_parentfield, target_d)
 
 	return target_d
+
+
+def set_taxes_and_charges(source_doc, target_doc):
+	if source_doc.get("customer") and source_doc.get("company"):
+		if frappe.get_cached_value("Customer", source_doc.get("customer"), "taxes_and_charges_template"):
+			from erpnext.accounts.party import set_taxes
+			from erpnext.accounts.doctype.sales_invoice.pos import update_tax_table
+			target_doc.taxes_and_charges = set_taxes(source_doc.get("customer"), "Customer", source_doc.get("posting_date"), source_doc.get("company"))
+			update_tax_table(target_doc)
```

Comment:
- Duplicated logic .. you have already implemented logic to determine taxes and charges based on customer and tax rules, do not reimplement it
- The idea is to use the same method that sets taxes and charges based on customer and tax rules
- Explore and use `TransactionController.reset_taxes_and_charges` in https://github.com/ParaLogicTech/erpnext/blob/1a46d40bd3675ba49cdafb768d4d967768431abd/erpnext/controllers/transaction_controller.py#L725
- For reference, this is already implemented in https://github.com/ParaLogicTech/erpnext/blob/1a46d40bd3675ba49cdafb768d4d967768431abd/erpnext/projects/doctype/project/project.py#L2130

revert everything